src/service.ts

```diff
@@ -136,38 +136,59 @@ function parseCLIArguments(args: string[]): [CLIArguments, string, CliOptions] {
 
   const optionArgs: string[] = [];
 
-  const argsIterator = args[Symbol.iterator]();
-  for (const arg of argsIterator) {
+  for (let i = 0; i < args.length; i++) {
+    let arg = args[i];
     if (arg.startsWith("-")) {
       switch (arg) {
         case "--no-color":
           parsedArguments.noColor = true;
           break;
 
         case "--ignore-path": {
-          const nextArg = argsIterator.next();
-          if (nextArg.done) {
+          const nextArg = args[i + 1];
+          if (nextArg === undefined) {
             throw new Error("--ignore-path option expects a file path");
           }
 
-          parsedArguments.ignorePath = nextArg.value;
+          parsedArguments.ignorePath = nextArg;
+          i++;
+          break;
+        }
+        case "--stdin-filepath": {
+          const nextArg = args[i + 1];
+          if (nextArg === undefined) {
+            throw new Error("--stdin-filepath option expects a file path");
+          }
+
+          fileName = nextArg;
+          i++;
           break;
         }
         default: {
-          optionArgs.push(arg);
+          if (arg.includes("=")) {
+            optionArgs.push(arg);
+          } else {
+            const nextArg = args[i + 1];
+            if (nextArg === undefined) {
+              throw new Error(`--${arg} expects a value`);
+            }
```

Comment:
do we still want this check?

Reply:
Thanks for pointing that out. Made me realise I missed some edge cases.